docs/recipes/react/example-forms.md

```diff
@@ -0,0 +1,192 @@
+---
+id: example-forms
+title: Example: Forms
+sidebar_label: Example: Forms
+---
+
+#### Example 1
+```js
+import React from 'react'
+import ReactDOM from 'react-dom'
+import {createEffect, createStore, createEvent, sample} from 'effector'
+import {useStore, useStoreMap} from 'effector-react'
+
+const sendForm = createEffect({handler: console.log})
+const submitted = createEvent()
+const setField = createEvent()
+const $form = createStore({}).on(setField, (s, {key, value}) => ({
+  ...s,
+  [key]: value,
+}))
+
+sample({
```

Comment:
How is `sample` better than
```js
const sendForm = createEffect(() =>
{
    console.log($form.getState());
}
```

and using `sendForm` directly, rather than triggering it via `submitted` event? In my example above we benefit from not creating excessive event (`submitted`)

Reply:
`getState` is more of an escape hatch,  neither the actual way of resolving store's state. As it was mentioned in recipe's code breakdown, `getState` may cause state inconsistency and race conditions. This code still works, and even though, mostly, you will be unable to observe this kind of effect, yet in certain situations this may be the real pain in the neck.

Reply:
Thank you! Appreciate the response